failsafe/circuit_breaker.py

```diff
@@ -57,9 +87,17 @@ def allows_execution(self):
         return True
 
     def record_success(self):
+        """
+        Sets the current number of failures to 0.
+        """
         self.current_failures = 0
 
     def record_failure(self):
+        """
+        Adds a failure to the state and if the number of failures
+        has reached the allowed number of failures, changes the state
```

Comment:
I would say:

...if the number of ** consecutive** failures has reached the allowed number...

It is important to understand that failures need to come after each other.